lib/Twig/Extension/Core.php

```diff
@@ -1277,7 +1277,15 @@ function twig_lower_filter(Twig_Environment $env, $string)
     function twig_title_string_filter(Twig_Environment $env, $string)
     {
         if (null !== ($charset = $env->getCharset())) {
-            return mb_convert_case($string, MB_CASE_TITLE, $charset);
+            $title = mb_convert_case($string, MB_CASE_TITLE, $charset);
+            $title = preg_replace('#\'S#u', '\'s', $title);
+
+            return preg_replace_callback(
+                            '#\b\d+([^\d\s]+)\b#u',
+                            function ($matches) {
```

Comment:
I should've know that since I even sent a PR to push to 5.2.7 :/
I'll see if can refactor this into something useful that works for 5.2 as well. Let me know if you've an idea about how to solve this ticket, maybe a whole different approach would be better.